src/modelmanager.cpp

```diff
@@ -913,17 +913,25 @@ Status ModelManager::loadConfig(const std::string& jsonFilename) {
         intermediateStatus = StatusCode::OK;
         break;
     } while (++counter < MAX_CONFIG_JSON_READ_RETRY_COUNT && !intermediateStatus.ok());
+    lastConfigFileMD5 = md5;
     if (!intermediateStatus.ok()) {
         this->lastLoadConfigStatus = intermediateStatus;
         return this->lastLoadConfigStatus;
     }
+    return StatusCode::OK;
+}
 
+Status ModelManager::loadConfig(const std::string& jsonFilename) {
+    rapidjson::Document configJson;
+    Status status = parseConfig(jsonFilename, configJson);
+    if (!status.ok())
+        return status;
+    std::lock_guard<std::recursive_mutex> loadingLock(configMtx);
```

Comment:
Why do we need to lock twice now? It introduces small time frame where lock is free. Is there some reason why you changed single lock at the beginning of loadConfig to multiple locks?